src/txdb.cpp

```diff
@@ -448,6 +450,47 @@ bool CBlockTreeDB::WriteTxIndex(const std::vector<std::pair<uint256, CTxIndexVal
     return WriteBatch(batch);
 }
 
+bool CBlockTreeDB::ReadMaturityHeightIndex(const int height, std::vector<CMaturityHeightKey> &val) {
+    boost::scoped_ptr<leveldb::Iterator> pcursor(NewIterator());
+
+    CDataStream ssKeySet(SER_DISK, CLIENT_VERSION);
+    ssKeySet << make_pair(DB_MATURITY_HEIGHT, CMaturityHeightIteratorKey(height));
+    pcursor->Seek(ssKeySet.str());
+
+    while (pcursor->Valid()) {
+        boost::this_thread::interruption_point();
+        try {
+            leveldb::Slice slKey = pcursor->key();
+            CDataStream ssKey(slKey.data(), slKey.data()+slKey.size(), SER_DISK, CLIENT_VERSION);
```

Comment:
If performance becomes an issue,

We could try to persist
- key: height
- value: List of mature certificates at heigh specified in key

Not sure how Slice works and if this is as performant as the solution proposed.

Reply:
The drawback of this approach is that several updates are going to be performed on the same object implying deserializing, serializing and saving the whole list each time (consider that epoch lengths could be different between sidechains and different blocks can trigger the modification of the same maturityheight).

Reply:
alberto's comment is related to the rdoria's comment.